src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -5748,136 +5754,144 @@ void func_808355D8(PlayState* play, Player* this, PlayerAnimationHeader* anim) {
     func_8082E1F0(this, NA_SE_IT_DEKUNUTS_FLOWER_CLOSE);
 }
 
-// related to grottos (?)
-s32 func_8083562C(PlayState* play, Player* this, CollisionPoly* poly, s32 bgId) {
-    s32 var_a3;
+s32 Player_HandleExitsAndVoids(PlayState* play, Player* this, CollisionPoly* poly, s32 bgId) {
+    s32 exitIndexPlusOne;
     FloorType floorType;
     s32 sp34;
     s32 sp30;
 
     if ((this == GET_PLAYER(play)) && !(this->stateFlags1 & PLAYER_STATE1_80) && !func_8082DA90(play) &&
         (this->csMode == PLAYER_CSMODE_0) && !(this->stateFlags1 & PLAYER_STATE1_1)) {
-        var_a3 = 0;
-        if (((poly != NULL) && (var_a3 = SurfaceType_GetSceneExitIndex(&play->colCtx, poly, bgId), (var_a3 != 0)) &&
-             (((play->sceneId != SCENE_GORONRACE) && (play->sceneId != SCENE_DEKU_KING)) || (var_a3 < 3)) &&
-             (((play->sceneId != SCENE_20SICHITAI) && (play->sceneId != SCENE_20SICHITAI2)) || (var_a3 < 0x15)) &&
-             ((play->sceneId != SCENE_11GORONNOSATO) || (var_a3 < 6))) ||
-            (func_808340D4(sPlayerCurrentFloorType) && (this->floorProperty == FLOOR_PROPERTY_12))) {
+        exitIndexPlusOne = 0;
+
+        if (((poly != NULL) &&
+             (exitIndexPlusOne = SurfaceType_GetSceneExitIndex(&play->colCtx, poly, bgId), (exitIndexPlusOne != 0)) &&
```

Comment:
:despair:
Does this mean Surfaces store sceneExitIndices + 1, not just the exit index directly?

Reply:
If you look at `play->nextEntrance = play->setupExitList[exitIndex];`, then yes. `SurfaceType_GetSceneExitIndex` will return the value 1 above that number because of `func_808354A4(play, exitIndexPlusOne - 1, ...)`

Reply:
Added a comment to `SurfaceType_GetSceneExitIndex` mentioning this